src/constructs/loadbalancing/clb.ts

```diff
@@ -48,3 +54,34 @@ export class GuClassicLoadBalancer extends LoadBalancer {
       Object.entries(mergedProps.propertiesToOverride).forEach(([key, value]) => cfnLb.addPropertyOverride(key, value));
   }
 }
+
+interface GuHttpsClassicLoadBalancerProps extends Omit<GuClassicLoadBalancerProps, "listeners"> {
+  listener?: Partial<LoadBalancerListener>;
+}
+
+export class GuHttpsClassicLoadBalancer extends GuClassicLoadBalancer {
+  static DefaultListener: LoadBalancerListener = {
+    internalPort: 9000,
+    externalPort: 443,
+    internalProtocol: LoadBalancingProtocol.HTTP,
+    externalProtocol: LoadBalancingProtocol.HTTPS,
+  };
+
+  constructor(scope: GuStack, id: string, props: GuHttpsClassicLoadBalancerProps) {
+    const listenerProps = { ...GuHttpsClassicLoadBalancer.DefaultListener, ...props.listener };
+
+    if (!listenerProps.sslCertificateId) {
+      const certificateId = new GuArnParameter(scope, "CertificateARN", {
```

Comment:
ACM certificates can be created via CloudFormation. Historically, it's been painful as it's a blocking resource - the stack will remain in "updating" until the certificate has been validated.

Validation takes two forms:
- If email validated, an email will be sent with a link to be clicked. This email goes to a list of domain admins (around 5 people)
- If DNS validated, a CNAME needs created in DNS. Again, admins for the domain service is around 5 people.

However, since the creation of https://github.com/guardian/dns-validation-lambda, it's become more viable to create ACM certs via CFN if domain validated.

(almost at the end of this story!)

Rather than this construct bringing a param for the user to fill in at runtime, should/could it bring an ACM resource with it? Providing the benefit of:
- Less involvement for user (aka reduce human error)
- Applies best practice of domain validated certificates (best practice because they auto-renew vs. email validated which needs manual renewal once a year)

Reply:
That would be cool (and thanks for providing the context)! I guess if we find that not all cases can move to this for whatever reason then we can always provide a prop to add a param instead. I'll add this as an idea to the discussions section for now. 💡